crates/bevy_time/src/common_conditions.rs

```diff
@@ -97,11 +97,11 @@ pub fn on_fixed_timer(duration: Duration) -> impl FnMut(Res<FixedTime>) -> bool
 /// }
 /// ```
 pub fn time_passed_since_app_started(
-    seconds: f32,
+    duration: Duration,
 ) -> impl FnMut(Local<f32>, Res<Time>) -> bool + Clone {
     move |mut timer: Local<f32>, time: Res<Time>| {
         *timer += time.delta_seconds();
-        *timer >= seconds
+        *timer >= duration.as_secs_f32()
     }
```

Comment:
I believe this can be simplified using [`Time::elapsed`](https://docs.rs/bevy/latest/bevy/time/struct.Time.html#method.elapsed), which should give you the `Duration` since startup directly. Then we can compare this to `duration` and don't need the `Local<f32>` or conversions to seconds :)